pkg/asset/manifests/template.go

```diff
@@ -1,5 +1,31 @@
 package manifests
 
+// AWS holds encoded credentials and is used to generate cloud-creds-secret
+type AWS struct {
+	Base64encodeAccessKeyID     string
+	Base64encodeSecretAccessKey string
+}
+
+// OpenStack holds encoded credentials and is used to generate cloud-creds-secret
+// TODO: Fill this
+type OpenStack struct {
+	Base64encodeSomething     string
+	Base64encodeSomethingElse string
+}
+
+// Libvirt holds encoded credentials and is used to generate cloud-creds-secret
+// TODO: Fill this
+type Libvirt struct {
+	Base64encodeSomething     string
+	Base64encodeSomethingElse string
+}
+
+type cloudCredsTemplateData struct {
+	AWS
```

Comment:
Explicitly name these fields. You are not accessing the fields directly from the `cloudCredsTemplateData` struct, so there is no reason to promote the members of the fields to the `cloudCredsTemplateData` struct.

Also, the fields should be pointers since they are optional.

```
type cloudCredsTemplateData struct {
   AWS *AWSCredsTemplateData
   OpenStack *OpenStackCredsTemplateData
   Libvirt *LibvirtCredsTemplateData
}
```

Reply:
I was going to ask you about that yesterday, thanks, I thought I was missing something there